pyresample/geometry.py

```diff
@@ -958,21 +967,39 @@ def freeze(self, lonslats=None, resolution=None, shape=None, proj_info=None):
         shape = None if None in shape else shape
         area_extent = self.area_extent
         if not area_extent or not width or not height:
-            proj4 = Proj(proj_dict)
-            try:
-                lons, lats = lonslats
-            except (TypeError, ValueError):
-                lons, lats = lonslats.get_lonlats()
-            xarr, yarr = proj4(np.asarray(lons), np.asarray(lats))
-            xarr[xarr > 9e29] = np.nan
-            yarr[yarr > 9e29] = np.nan
-            corners = [np.nanmin(xarr), np.nanmin(yarr),
-                       np.nanmax(xarr), np.nanmax(yarr)]
+            corners = self._compute_corners(proj_dict, lonslats)
             area_extent, width, height = self.compute_domain(corners, resolution, shape)
         return AreaDefinition(self.area_id, self.description, '',
                               projection, width, height,
                               area_extent, self.rotation)
 
+    def _compute_corners(self, proj_dict, lonslats):
+        proj4 = Proj(proj_dict)
+        lons, lats = self._extract_lons_lats(lonslats)
+        # TODO: Do more dask-friendly things here
```

Comment:
Please make an issue about this. This will get lost in the code otherwise...

Reply:
#344